### PR TITLE
ARX-7 Refactor

### LINKED ISSUE #39: Create an IRC Admin Interface
Aoi-chan accidentally put the Imgur plugin in for Commie, so Rhe got mad at me again. He suggested adding an IRC interface / recognized commands to modify the bot while it's running. Adding / Removing plugins shouldn't be too difficult.

### PR DESCRIPTION
**Pull request by IanMitchell**
Much easier to test / tweak now.

Fixes #39

Starts work on #34
Enables work on #40

### CODE REVIEW

File: src/arx7.js [hunk 1]
```diff
@@ -1,133 +1,199 @@
...
+      log(`Unrecognized command`);
+      this.client.say(from, `Command not recognized, boss. ${usage}`);
       return;
     }
 
-    let key = config.channels[channels.indexOf(message.args[1])].key;
+    // Verify command format
+    if (args.length != 4) {
+      log(`Unrecognized command`);
+      this.client.say(from, `Not enough commands. ${usage}`);
```
Comment (kunaldes) [inline src/arx7.js]: Maybe "incorrect number of arguments"? because it could have more than 4 and still trigger this case.

File: src/arx7.js [hunk 2]
```diff
@@ -1,133 +1,199 @@
...
-    if (droppedChannels.includes(message.args[1])) {
-      log(`Incorrect password for ${message.args[1]}.`);
+
+  handleAdminQuery(from, to, text, message) {
+    log(`Admin Query from ${from}: ${text}`);
+    let usage = `[add|remove] [plugin] [channel] [password]`;
+    let args = text.split(/\s+/);
+
+    // We only support add/remove commands
+    if (!(text.startsWith('add ') || text.startsWith('remove '))) {
```
Comment (kunaldes) [inline src/arx7.js]: you call .toLowerCase() below, but this check is case sensitive, right?

File: src/arx7.js [hunk 3]
```diff
@@ -1,133 +1,199 @@
...
+    if (message.command == 'err_bannedfromchan') {
+      log(`Banned from ${message.args[1]}. Rejoining in in 3 minutes`);
+
+      // `()=>` ensures there is a delay; otherwise it continuously fires
+      setTimeout(() => this.client.join(message.args[1]), 1000 * 60 * 3);
+    }
+    // Attempt to rejoin +k channels correctly
+    else if (message.command == 'err_badchannelkey') {
+      // Prevent infinite rejoin attempts
+      if (this.droppedChannels.includes(message.args[1])) {
```
Comment (kunaldes) [inline src/arx7.js]: Not a new bug, but I just realized if you kick the bot from a channel with +k twice, it won't join the second time. Is it possible to remove a channel from the list when the bot successfully joins a channel?

File: src/arx7.js (hunk 1 above)
Comment (IanMitchell) [inline src/arx7.js]: Yeah, that's better

File: src/arx7.js (hunk 2 above)
Comment (IanMitchell) [inline src/arx7.js]: Yeah, good catch

File: src/arx7.js (hunk 3 above)
Comment (IanMitchell) [inline src/arx7.js]: Oh, yeah. Nice find

**Pull request merged**